mmdet/datasets/pipelines/loading.py

```diff
@@ -625,6 +625,8 @@ def __call__(self, results):
         for t in tests[1:]:
             keep = keep & t
 
+        keep = keep.nonzero()[0]
```

Comment:
There is something wrong in the unit test, please check it

![image](https://user-images.githubusercontent.com/26483343/172103656-f6022ad3-33b6-45c8-8b9c-4a504b3dadae.png)

Reply:
since we changed the filtering type from mask to indexes, we can no longer use ```keep.any()``` to check, because it does not work at index zero.
But we can use

```
if keep.size == 0:
             if self.keep_empty:
                 return None
```

Reply:
I tested this function locally with unit tests, it works for me